communicator/ssh/communicator.go

```diff
@@ -236,7 +298,7 @@ func (c *comm) Download(path string, output io.Writer) error {
 		return nil
 	}
 
-	return c.scpSession("scp -vf "+strconv.Quote(path), scpFunc)
+	return c.scpSession("scp -vf "+path, scpFunc)
```

Comment:
Oh I see. So if we quote the string the * doesn't get interpreted as a glob expansion.

How does this currently work with upload? I *think* it just does the whole file without `*`. For example if you want everything under `/some/path/*` you simply say `/some/path`.

I'm curious if removing `Quotes()` breaks backwards-compatibility in some way. I think it's unlikely that someone would use a * in a filename but I'm sure spaces are common, especially on Windows. Is there a way we can preserve backwards-compatibility in this case? For example, can we call `Quotes()` if we detect that the string contains a space and no *?

Some examples -- I'm not sure if all of these work currently, though:

```
"/path/to/some folder/file"
"/path/to/some file"
"/path/to/some\ file"
"C:\path\to\some file"
"C:\path\to\some\\ file"
"/path/to/some\ folder/*"
"/path/to/some*folder/*"
"/path/to/some folder/*"   // Needs the space escaped because of glob
```